index.js

```diff
@@ -65,6 +83,16 @@ class S3Adapter {
       globalCacheControl: this._globalCacheControl,
     };
 
+    if (options.accessKey && options.secretKey) {
+      awsCredentialsDeprecationNotice();
+      s3Options.credentials = {
+        accessKeyId: options.accessKey,
+        secretAccessKey: options.secretKey,
+      };
+    } else if (options.credentials) {
+      s3Options.credentials = options.credentials;
```

Comment:
Do we need a test for this?

Reply:
Yes we need a test for this part and i added it.